use ChainDB and Shelley node-to-node client/server protocols

avieth opened the pull request.

avieth committed:
download to ChainDB

This is a massive commit because it comes from branch work on
ourobors-network, but byron-proxy was migrated to this repo in the
interim.

avieth committed:
new-style Shelley node-to-node server

avieth committed:
update codecs and dependencies

mrBliss committed:
Use a better diskPolicy for the LedgerDB

See the comment for more information. This significantly speeds up startup
with a non-empty database.

avieth committed:
Shelley consumer side implemented

Can download from a cardano-node using node-to-node initiator side

avieth committed:
remove useless lines in stack.yaml

avieth committed:
remove unused imports from Main

dcoutts committed:
Sync the cabal.project file so it builds

avieth committed:
undo accidentally-commited change to regenerate.sh

avieth committed:
update ouroboros-network rev and fallout

avieth committed:
update cabal.project

avieth committed:
fixes after rebase

avieth committed:
apply nix patch (locally wouldn't do it right)

avieth committed:
fix nix problems (hopefully)

updated nix src file

avieth committed:
bump cardano-shell

avieth committed:
local tx submission codec is error

See note inline

avieth committed:
add cardano-sl config and mainnet-genesis

Made the new-style cardano-ledger config also look in the configuration
dir, like the cardano-sl one does.

When loading the configuration, the genesis json (if needed) is sourced
from the configuration directory.

avieth committed:
remove old unused file

avieth committed:
fix bug in sqlite index lookup

also make the insert fail on conflict

avieth committed:
announce new tip when stable for 1s or more

A good enough approach.

avieth committed:
another SQlite index fix (same problem)

mrBliss commented on src/exec/Shelley.hs:
```diff
@@ -121,9 +122,11 @@ codecs = ProtocolCodecs
-      -- Copied from cardano-node's choice.
-      Serialise.encode
-      Serialise.decode
+      -- There is at the moment no codec for UTxOValidationError
```
My fault, but I'm working on it: https://github.com/input-output-hk/cardano-ledger/pull/598. When that has been merged, I still need to incorporate it in `ouroboros-network`, sigh.

@coot In a case like this it would be nice to make the node-to-client protocols optional.

cleverca22 committed:
fix connect scripts
add a clean-source for the mainnet state
log every insertion into the sqlite index

avieth committed:
sqlite index tracing

avieth committed:
bump cardano-ledger, slot duration change

It's now a Natural, not NominalDiffTime. This is good: NominalDiffTime
allowed sub-milliseconds to be expressed, but Byron can't handle that.

avieth committed:
local tx submission codec defined

requires a new revision of cardano-ledger for the validation error
to/from CBOR instances

avieth committed:
Merge pull request #10 from input-output-hk/cleverca22/chaindb

fix a number of things

avieth committed:
add imports of toCBOR fromCBOR

accidentally removed in a merge

cleverca22 committed:
fix build errors

avieth committed:
move genesis conversion stuff from Main.hs

avieth committed:
improve host/port parser

It uses [hostname]:port format

avieth committed:
bump ouroboros-network, cardano-ledger

avieth committed:
cardano-ledger revision is in master

avieth committed:
remove validator scripts from nix

The validator executable is long dead. cardano-byron-proxy itself does
validation by way of the ChainDB

cleverca22 committed:
regen nix files
set stack.yaml to the tracer version of ouroboros-network
remove validator tests

avieth committed:
bump cardano-ledger again

avieth committed:
Merge pull request #11 from input-output-hk/cleverca22/chaindb

fix build errors

avieth committed:
remove redundant import

avieth committed:
remove redundant import

avieth committed:
byron download exception handler

avieth committed:
bump ouroboros-network

avieth commented:
bors r+

avieth commented:
@dcoutts agrees that this branch is basically master. Thanks @cleverca22 for the approval.

avieth commented:
bors r+
